tests/socket/socket_tests.cpp

```diff
@@ -156,6 +156,109 @@ connection_test(
     receiver_socket.complete_async_receive();
 }
 
+void
+auth_connection_test(
```

Comment:
This must test for contents of `sub_interface_map` and the other maps populated via the new helpers.

Reply:
Done.